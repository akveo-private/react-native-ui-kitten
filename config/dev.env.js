const path = require('path');

const evaPath = path.resolve(__dirname, '../../eva');

module.exports = {
  ENV: 'dev',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(evaPath, 'packages/mapping/eva'),
  THEME_PATH: path.resolve(evaPath, 'packages/theme/eva'),
};
