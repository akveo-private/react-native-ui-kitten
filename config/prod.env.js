const path = require('path');

const playgroundPath = path.resolve(__dirname, '../src/playground');

module.exports = {
  ENV: 'prod',
  KITTEN_PATH: path.resolve(__dirname, '../src/framework'),
  MAPPING_PATH: path.resolve(playgroundPath, 'node_modules/@eva/eva'),
  THEME_PATH: path.resolve(playgroundPath, 'node_modules/@eva/theme-eva'),
};
