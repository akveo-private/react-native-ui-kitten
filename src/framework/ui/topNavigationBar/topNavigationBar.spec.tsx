import React from 'react';
import {
  render,
  fireEvent,
  shallow,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  TopNavigationBar as TopNavigationBarComponent,
  Props as TopNavigationBarProps,
} from './topNavigationBar.component';
import {
  TopNavigationBarAction as TopNavigationBarActionComponent,
  Props as TopNavigationBaActionProps,
} from './topNavigationBarAction.component';
import * as config from './topNavigationBar.spec.config';
import { TouchableWithoutFeedback } from 'react-native';

const TopNavigationBar = styled<TopNavigationBarComponent, TopNavigationBarProps>(TopNavigationBarComponent);
const TopNavigationBarAction =
  styled<TopNavigationBarActionComponent, TopNavigationBaActionProps>(TopNavigationBarActionComponent);

const Mock = (props?: TopNavigationBarProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <TopNavigationBar {...props} />
  </StyleProvider>
);

const ActionMock = (props?: TopNavigationBaActionProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <TopNavigationBarAction {...props} />
  </StyleProvider>
);

const iconSourceUri: string = 'https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png';
const testIdLeftAction: string = '@top-navbar-left';
const testIdRightAction1: string = '@top-navbar-right-1';
const testIdRightAction2: string = '@top-navbar-right-2';
const testIdRightAction3: string = '@top-navbar-right-3';

describe('@top-navigation-bar/action', () => {

  it('* bar/title', () => {
    const { output } = shallow(<Mock title='Test'/>);
    expect(output).toMatchSnapshot();
  });

  it('* bar/subtitle', () => {
    const { output } = shallow(<Mock title='Test' subtitle='Subtitle'/>);
    expect(output).toMatchSnapshot();
  });

  it('* bar/appearance: title-centered', () => {
    const { output } = shallow(<Mock appearance='title-centered' title='Test'/>);
    expect(output).toMatchSnapshot();
  });

  it('* bar/with actions', () => {
    const onLeftControl = jest.fn();
    const onRightControl1 = jest.fn();
    const onRightControl2 = jest.fn();
    const onRightControl3 = jest.fn();

    const component =
      <Mock
        title='Test'
        subtitle='Subtitle'
        leftControl={
          <ActionMock testID={testIdLeftAction} iconSource={{ uri: iconSourceUri }} onPress={onLeftControl}/>
        }
        rightControls={[
          <ActionMock testID={testIdRightAction1} iconSource={{ uri: iconSourceUri }} onPress={onRightControl1}/>,
          <ActionMock testID={testIdRightAction2} iconSource={{ uri: iconSourceUri }} onPress={onRightControl2}/>,
          <ActionMock testID={testIdRightAction3} iconSource={{ uri: iconSourceUri }} onPress={onRightControl3}/>,
        ]}
      />;
    const renderedComponent = render(component);
    const { output } = shallow(component);

    expect(output).toMatchSnapshot();
    fireEvent.press(renderedComponent.getByTestId(testIdLeftAction));
    fireEvent.press(renderedComponent.getByTestId(testIdRightAction1));
    fireEvent.press(renderedComponent.getByTestId(testIdRightAction2));
    fireEvent.press(renderedComponent.getByTestId(testIdRightAction3));
    expect(onLeftControl).toHaveBeenCalled();
    expect(onRightControl1).toHaveBeenCalled();
    expect(onRightControl2).toHaveBeenCalled();
    expect(onRightControl3).toHaveBeenCalled();
  });

  it('* action/with icon uri', () => {
    const { output } = shallow(<ActionMock iconSource={{uri: iconSourceUri}}/>);
    expect(output).toMatchSnapshot();
  });

  it('* action/is last item check (true)', () => {
    const { output } = shallow(<ActionMock iconSource={{uri: iconSourceUri}} isLastItem={true}/>);
    expect(output).toMatchSnapshot();
  });

  it('* action/is last item check (false)', () => {
    const { output } = shallow(<ActionMock iconSource={{uri: iconSourceUri}} isLastItem={false}/>);
    expect(output).toMatchSnapshot();
  });

  it('* action/on press check', () => {
    const onPress = jest.fn();
    const component = <ActionMock iconSource={{uri: iconSourceUri}} onPress={onPress}/>;
    const renderedComponent = render(component);
    const { output } = shallow(component);

    fireEvent.press(renderedComponent.getByType(TouchableWithoutFeedback));
    expect(onPress).toHaveBeenCalled();
    expect(output).toMatchSnapshot();
  });

});
