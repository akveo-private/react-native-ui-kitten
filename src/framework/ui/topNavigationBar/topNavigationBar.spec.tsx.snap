// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@top-navigation-bar/action * action/is last item check (false) 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    iconSource={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    isLastItem={false}
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * action/is last item check (true) 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    iconSource={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    isLastItem={true}
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * action/on press check 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    iconSource={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    onPress={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * action/with icon uri 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    iconSource={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * bar/appearance: title-centered 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    appearance="title-centered"
    title="Test"
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * bar/subtitle 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    subtitle="Subtitle"
    title="Test"
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * bar/title 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    title="Test"
  />
</StyleProvider>
`;

exports[`@top-navigation-bar/action * bar/with actions 1`] = `
<StyleProvider
  mapping={
    Object {
      "TopNavigationBar": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "backgroundColor": "blue-primary",
              "height": 46,
              "paddingBottom": 12,
              "paddingHorizontal": 16,
              "paddingTop": 4,
              "subtitle.color": "white",
              "subtitle.fontSize": 12,
              "subtitle.fontWeight": "400",
              "title.centered": false,
              "title.color": "white",
              "title.fontSize": 16,
              "title.fontWeight": "600",
            },
          },
          "title-centered": Object {
            "mapping": Object {
              "title.centered": true,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
      "TopNavigationBarAction": Object {
        "appearance": Object {
          "default": Object {
            "mapping": Object {
              "height": 25,
              "marginRight": 8,
              "width": 25,
            },
          },
        },
        "meta": Object {
          "states": Array [],
          "variants": Object {},
        },
      },
    }
  }
  styles={Object {}}
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
>
  <ForwardRef(createComponent)
    leftControl={
      <ActionMock
        iconSource={
          Object {
            "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
          }
        }
        onPress={[MockFunction]}
        testID="@top-navbar-left"
      />
    }
    rightControls={
      Array [
        <ActionMock
          iconSource={
            Object {
              "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
            }
          }
          onPress={[MockFunction]}
          testID="@top-navbar-right-1"
        />,
        <ActionMock
          iconSource={
            Object {
              "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
            }
          }
          onPress={[MockFunction]}
          testID="@top-navbar-right-2"
        />,
        <ActionMock
          iconSource={
            Object {
              "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
            }
          }
          onPress={[MockFunction]}
          testID="@top-navbar-right-3"
        />,
      ]
    }
    subtitle="Subtitle"
    title="Test"
  />
</StyleProvider>
`;
